dev/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cpp

```diff
@@ -7,15 +7,32 @@
 #include "RuntimeProfiler.h"
 #include "ResourceAccessor.h"
 
+winrt::IMap<winrt::hstring, winrt::RadioMenuFlyoutItem> RadioMenuFlyoutItem::s_selectionMap = nullptr;
+
 RadioMenuFlyoutItem::RadioMenuFlyoutItem()
 {
     __RP_Marker_ClassById(RuntimeProfiler::ProfId_RadioMenuFlyoutItem);
 
     m_InternalIsCheckedChangedRevoker = RegisterPropertyChanged(*this, winrt::ToggleMenuFlyoutItem::IsCheckedProperty(), { this, &RadioMenuFlyoutItem::OnInternalIsCheckedChanged });
 
+    if (!s_selectionMap)
+    {
+        // Ensure that this object exists
+        s_selectionMap = winrt::single_threaded_map<winrt::hstring, winrt::RadioMenuFlyoutItem>();
+    }
+
     SetDefaultStyleKey(this);
 }
 
+RadioMenuFlyoutItem::~RadioMenuFlyoutItem()
+{
+    // If this is the checked item, remove it from the lookup.
+    if (IsChecked())
+    {
+        s_selectionMap.Insert(GroupName(), nullptr);
```

Comment:
as an aside... anyone know why this never gets hit?

Reply:
@jevansaks @StephenLPeters any ideas ?

Reply:
I was getting the destructor hit sometimes before, but now I'm not even getting that. I assumed that leaving the page would trigger it... is there something else I need to hook it up?

Reply:
If it's a managed app, you're at the mercy of the GC. I usually find if you navigate back and then to a different page it'll get cleaned up. But for more deterministic behavior you can add a case to the CX test app.